src/SeriesTracker.js

```diff
@@ -10,65 +10,76 @@ export function SeriesTracker(series) {
+      } else {
+        // Converting from DD.MM.YYYY to MM.DD.YYYY format and comparing finishedDate of serie and "lastSerie" with Date() object for a more accurate result 
+        let lastSerieArray = this.lastSerie.finishedDate.split(".");
+        [lastSerieArray[0], lastSerieArray[1]] = [lastSerieArray[1], lastSerieArray[0]];
+        let serieArray = serie.finishedDate.split(".");
+        [serieArray[0], serieArray[1]] = [serieArray[1], serieArray[0]];
+        const lastSerieTempDate = new Date(lastSerieArray);
+        const serieTempDate = new Date(serieArray);
+        if (serieTempDate > lastSerieTempDate) {
+          this.lastSerie = serie;
+        }
+      }
     } else {
-      // If a serie hasn't been watched:
-      // Check if serie has "isCurrent" prop
-      // Check if we have a "currentSerie" property. Set if we don't.
-      // Check if we have a "nextSerie" property as well - if we didn't
-      // set the .currentSerie property, set the .nextSerie property.
+      this.numberOfUnWatched += 1;
+      // Checking "isCurrent" property in serie and setting it to "currentSerie" if not.
+      if ("isCurrent" in serie && this.currentSerie === undefined) {
+        this.currentSerie = serie;
+      } else if (this.nextSerie === undefined) {
+        this.nextSerie = serie;
+      }
     }
-
-    //it should also update the number of series marked as watched / unwatched:
-    //"numberOfWatched" and "numberOfUnWatched"
   };
 
   //check to see if we have series to process
   if (series.length > 0) {
-    //Loop through all of the series in the "series" argument
-    //Use the .add function to handle adding series, so we keep counts updated.
+    series.forEach(serie => {
+      this.add(serie);
+    })
   }
 
   this.finishSerie = function () {
-    // find and update currently watching serie in "this.series" array
-    // update "lastSerie" with the finished one
-    // set "currentSerie" with the next one
-    // set new nextSerie value with the next one which has not been watched.
-    // update "numberOfWatched" and "numberOfUnWatched" props
-  };
-  this.printSeriesReport = function () {
-    fancyLogSeriesReport(this);
-  };
-}
+    this.numberOfWatched += 1;
+    this.numberOfUnWatched -= 1;
 
-// Case 1
-// -------------------------------------------------
+    // Updating "lastSerie" with the finished one and "currentSerie" with the next unwatched one
+    this.lastSerie = this.currentSerie;
+    this.currentSerie = this.nextSerie;
 
-/* const mySeriesTracker = new SeriesTracker(series);
-mySeriesTracker.printSeriesReport(); */
+    // Getting today's date with Date object
+    const date = new Date();
+    const today = ("0" + date.getDate()).slice(-2) + '.' + ("0" + (date.getMonth() + 1)).slice(-2) + '.' + date.getFullYear();
 
-// Case 2
-// -------------------------------------------------
+    // Updating "lastSerie" properties
+    const lastSerieIndex = this.series.findIndex((serie) => {
```

Comment:
Bu ve 70. satırdaki findIndexler yerine tek bir map ile tek bir döngü ve içerde iki if ile şartları sağlayan elemanları update edilmesi hem tek döngü olması hem daha az satır ve okunabilirlik için iyi olabilirdi.

Reply:
Teşekkürler.